packages/bus/src/Brokers/RMQEventBus.ts

```diff
@@ -0,0 +1,395 @@
+    }
+
+    this.channel = await this.client.createConfirmChannel();
+
+    await this.bindExchangesToQueue();
+    await this.startBasicConsume();
+    await this.startReplyQueueConsume();
+  }
+
+  private async startReplyQueueConsume(): Promise<void> {
+    if (!this.channel) {
+      return;
+    }
+
+    const { consumerTag } = await this.channel.consume(
+      this.REPLY_QUEUE_NAME,
+      (msg: ConsumeMessage | null) => this.processReply(msg),
+      {
+        noAck: true
+      }
+    );
+
+    this.consumerTags.push(consumerTag);
+  }
+
+  private async startBasicConsume(): Promise<void> {
+    if (!this.channel) {
+      return;
+    }
+
+    const { consumerTag } = await this.channel.consume(
+      this.configuration.clientQueue,
+      (msg: ConsumeMessage | null) => this.consumeReceived(msg),
+      {
+        noAck: true
+      }
+    );
+
+    this.consumerTags.push(consumerTag);
+  }
+
+  private async bindExchangesToQueue(): Promise<void> {
+    await this.channel?.assertExchange(this.configuration.exchange, 'direct', {
+      durable: true
+    });
+
+    await this.channel?.assertQueue(this.configuration.clientQueue, {
+      durable: true,
+      exclusive: false,
+      autoDelete: true
+    });
+  }
+
+  private processReply(message: ConsumeMessage | null): void {
+    const event: ParsedConsumeMessage | undefined =
+      this.parseConsumeMessage(message);
+
+    if (event && event.correlationId) {
+      this.subject.emit(event.correlationId, event.payload);
+    }
+  }
+
+  private async consumeReceived(message: ConsumeMessage | null): Promise<void> {
+    const event: ParsedConsumeMessage | undefined =
+      this.parseConsumeMessage(message);
+
+    if (!event) {
+      return;
+    }
+
+    const handlers = this.handlers.get(event.name);
+
+    if (!handlers) {
+      return;
+    }
+
+    for (const handler of handlers) {
+      await this.provideHandler(handler, event);
+    }
+  }
+
+  private async provideHandler(
+    handler: EventHandler<unknown, unknown>,
+    event: ParsedConsumeMessage
+  ): Promise<void> {
+    try {
+      const response = await handler.handle(event.payload);
+
+      if (!event.replyTo) {
+        return;
+      }
+
+      this.channel?.sendToQueue(
+        event.replyTo,
+        Buffer.from(JSON.stringify(response)),
+        {
+          mandatory: true,
+          correlationId: event.correlationId
+        }
+      );
+    } catch {
+      // TODO: add logs
+    }
+  }
+
+  private parseConsumeMessage(
+    message: ConsumeMessage | null
+  ): ParsedConsumeMessage | undefined {
+    if (!message) {
+      return;
+    }
+
+    if (message.fields.redelivered) {
+      return;
+    }
+
+    const { content, fields, properties } = message;
+    const { type, correlationId, replyTo } = properties;
+    const { routingKey } = fields;
+
+    const name = type ?? routingKey;
+
+    const payload: Event<any> = JSON.parse(content.toString());
```

Comment:
`payload` has nothing to do with `Event`

```suggestion
    const payload: unknown = JSON.parse(content.toString());
```